src/mpi-base.jl

```diff
@@ -525,6 +525,62 @@ function Reduce{T}(object::T, op::Op, root::Integer, comm::Comm)
     isroot ? recvbuf[1] : nothing
 end
 
+# T and T2 are different for MPI_IN_PLACE as the sendbuf
+function Allreduce{T}(sendbuf::MPIBuffertype{T}, recvbuf::MPIBuffertype{T},
+                      count::Integer, op::Op, comm::Comm)
+    flag = zero(Cint)
```

Comment:
Should be `Ref{Cint}()` in the Julia 0.4 style, with a `Ref{Cint}` argument.   In general, `&` is deprecated these days.